docs/dev/index.md

```diff
@@ -0,0 +1,68 @@
+
+# OpenSearch SQL/PPL Engine Development Manual
+
+## Introduction
```

Comment:
sure, I'm still thinking what's the best way to do this. I have a sub task here for adding tutorial / quick start section: https://github.com/opensearch-project/sql/issues/1219.

Many other projects provide sample data and executable script/jar. User can play around with sample data files via CLI and their file connector. Because our CLI is written in Python, probably we can put some quick start example for playground website.